src/App.tsx

```diff
@@ -1,78 +1,40 @@
-/**
- * Sample React Native App
- * https://github.com/facebook/react-native
- *
- * Generated with the TypeScript template
- * https://github.com/react-native-community/react-native-template-typescript
- *
- * @format
- */
+import React, { useState, useEffect } from 'react';
+import { View, Text, StyleSheet } from 'react-native';
+import { observer } from 'mobx-react-lite';
 
-import React, {useState, useEffect} from 'react';
-import {
-  SafeAreaView,
-  StyleSheet,
-  ScrollView,
-  View,
-  Text,
-  StatusBar,
-} from 'react-native';
-
-import {
-  Header,
-  LearnMoreLinks,
-  Colors,
-  DebugInstructions,
-  ReloadInstructions,
-} from 'react-native/Libraries/NewAppScreen';
+import { firebaseService } from './services';
+import { useStores } from './hooks';
 
+import Chat from './components/Chat';
 import Button from './components/common/Button';
-import { AppConfig } from './AppConfig';
-
-import auth, { firebase } from '@react-native-firebase/auth';
-//import styles from './components/Input/styles';
-
-declare var global: {HermesInternal: null | {}};
 
-const App = () => {
+const App = observer(() => {
+  const { userStore } = useStores()
   const [initializing, setInitializing] = useState(true);
-  const [user, setUser] = useState();
-  const [firebaseApp, setFirebaseApp] = useState();
 
   function onAuthStateChanged(user: any): void {
-    setUser(user);
+    userStore.setCurrentUser(user);
     if (initializing) setInitializing(false);
   }
 
-  function maybeInitializeFirebase(): void {
-    try {
-      if (!firebase.app()) {
-        firebase.initializeApp(AppConfig);
-      }
-    } catch (e) {
-      firebase.initializeApp(AppConfig);
-    }
-  }
-
   useEffect(() => {
-    maybeInitializeFirebase();
-    const subscriber = auth().onAuthStateChanged(onAuthStateChanged);
+    const subscriber = firebaseService.auth.onAuthStateChanged(onAuthStateChanged);
     return subscriber; // unsubscribe on unmount
   }, []);
 
   async function login(): Promise<void> {
     const email: string = "weilon+potato@weilonying.com";
     const password: string = "potatopotato";
     try {
-      await auth().signInWithEmailAndPassword(email, password);
+      await firebaseService.auth.signInWithEmailAndPassword(email, password);
     } catch (e) {
       console.error(e.message);
     }
   }
 
   async function logout(): Promise<void> {
     try {
-      await auth().signOut();
+      await firebaseService.auth.signOut();
     } catch (e) {
       console.error(e.message);
     }
```

Comment:
We should probably split App.tsx up into a main menu screen and a login/register screen.